src/python_tools/sensor_proto_log.py

```diff
@@ -0,0 +1,52 @@
+from .proto_log import ProtoLog
+from software.proto.sensor_msg_pb2 import SensorProto
+from google.protobuf.message import Message
+from typing import List, Dict
+
+
+class SensorProtoLog:
+    """
+    SensorProtoLog allows users to work with different types of messages 'multiplexed' into 'SensorProto'.
+    A list is created for each member message type of 'SensorProto', with the same name. All of the
+    messages of that type are put into the list for the given 'SensorProto' type data directory.
+
+    Usage:
+    ```
+    sensor_proto_log = SensorProtoLog('/tmp/test/SensorProto')
+    sensor_proto_log.ssl_vision_msg[1000]
+    ```
+    This gets the 1000th SSL_WrapperPacket message in the series of SensorProtos, which corresponds to
+    the member `ssl_vision_msg`.
+    """
+
+    def __init__(self, directory: str):
+        """
+        Constructs a SensorProtoLog for the given data directory.
+        :param directory: A directory containing RepeatedAnyMsg delimited files containg SensorProto's
+        """
+        proto_log = ProtoLog(directory, SensorProto)
+        self._messages: Dict[str, List[Message]] = dict()
+        for sensor_msg in proto_log:
+            # ListFields() gives a list of tuple of (FieldDescriptor, Message)
+            field_names = [field[0].name for field in sensor_msg.ListFields()]
+
+            # iterate over field names for each message and add them to the dict entry
+            # for that particular field
+            for field_name in field_names:
+                if field_name != "backend_received_time":
```

Comment:
why do we ignore this field?

Reply:
yeah we should probably demux the timestamps too

Reply:
done